extensions/jdbc/jdbc-mariadb/deployment/src/main/java/io/quarkus/jdbc/mariadb/deployment/MariaDBJDBCReflections.java

```diff
@@ -12,18 +11,10 @@ void build(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
         //Not strictly necessary when using Agroal, as it also registers
         //any JDBC driver being configured explicitly through its configuration.
         //We register it for the sake of people not using Agroal.
-        final String driverName = "org.mariadb.jdbc.Driver";
-        reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, driverName));
+        reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, "org.mariadb.jdbc.Driver"));
 
-        //MariaDB's connection process requires reflective read to all fields of Options:
-        reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, "org.mariadb.jdbc.util.Options"));
-    }
-
-    @BuildStep
-    void runtimeInit(BuildProducer<RuntimeInitializedClassBuildItem> runtimeInitialized) {
-        //MastersSlavesListener starts threads in DynamicSizedSchedulerImpl which is disallowed during build time in GraalVM
-        runtimeInitialized
-                .produce(new RuntimeInitializedClassBuildItem(
-                        "org.mariadb.jdbc.internal.failover.impl.MastersReplicasListener"));
```

Comment:
This one is gone and I couldn't find its successor, same for `DynamicSizedSchedulerImpl`. But I don't think our tests cover such a scenario, so there might be missing something here.

Reply:
PS: I'd actually expect native image _building_ to fail if a class still spawns threads and that doesn't happen.